src/services/edusharing/logic/connector.js

```diff
@@ -0,0 +1,167 @@
+	getCookie() {
+		const cookieOptions = {
+			uri: `${process.env.ES_DOMAIN}/edu-sharing/rest/authentication/v1/validateSession`,
+			method: 'GET',
+			headers: EduSharingConnector.authorization,
+			resolveWithFullResponse: true,
+			json: true
+		};
+		return request(cookieOptions)
+			.then(result => {
+				if (
+					result.statusCode !== 200 ||
+					result.body.isValidLogin !== true
+				) {
+					throw Error('authentication error with edu sharing');
+				}
+				return result.headers['set-cookie'][0];
+			})
+			.catch(err => {
+				// eslint-disable-next-line no-console
+				console.error('error: ', err);
+			});
+	}
+	// gets access_token and refresh_token
+	getAuth() {
+		const oauthoptions = {
+			method: 'POST',
+			url: `${process.env.ES_DOMAIN}/edu-sharing/oauth2/token`,
+			headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+			body: `grant_type=${"password"}&client_id=${"eduApp"}&client_secret=${"secret"}&username=${process.env.ES_USER}&password=${process.env.ES_PASSWORD}`,
+			timeout: REQUEST_TIMEOUT
+		};
+		return request(oauthoptions).then(result => {
+			if (result) {
+				const parsedResult = JSON.parse(result);
+				return parsedResult.access_token;
+			}
+			console.error('Oauth failed');
+			return null;
+		});
+	}
+
+	checkEnv() {
+		return true
+		/* return (
+			process.env.ES_DOMAIN &&
+			process.env.ES_USER &&
+			process.env.ES_PASSWORD &&
+			process.env.ES_GRANT_TYPE &&
+			process.env.ES_OAUTH_SECRET &&
+			process.env.ES_CLIENT_ID
+		); */
+	}
+
+	async login() {
+		this.authorization = await this.getCookie();
+		this.accessToken = await this.getAuth();
+	}
+
+	isLoggedin() {
+		// returns false if cookie or accesstoken is falsy
+		return !!this.authorization && !!this.accessToken;
+	}
+
+	async GET(data) {
+		const limit = data.query.$limit || 9;
+		const skip = data.query.$skip || 0;
+		const searchWord = data.query.searchWord || 'img'; // will give pictures of flowers as default
+		const contentType = data.query.contentType || 'FILES'; // enum: ['FILES', FILES_AND_FOLDERS,COLLECTIONS, TOOLPERMISSIONS, ALL]
+
+		if (!this.checkEnv()) {
+			return 'Update your env variables. See --> src/services/edusharing/envTemplate';
+		}
+
+		if (this.isLoggedin() === false) {
+			await this.login();
+		}
+
+		const options = {
+			method: 'POST',
+			// This will be changed later with a qs where sorting, filtering etc is present. 
+			// hard coded params
+			url: `https://mv-repo.schul-cloud.org/edu-sharing/rest/search/v1/queriesV2/-home-/-default-/workspace/?contentType=ALL&skipCount=0&maxItems=25&sortProperties=cm%3Aname&sortAscending=true&propertyFilter=-all-&`,
+			//url: `${process.env.ES_DOMAIN}/edu-sharing/rest/search/v1/queriesV2/mv-repo.schul-cloud.org/mds/ngsearch/?contentType=${contentType}
+			//&skipCount=${skip}&maxItems=${limit}&sortProperties=score&sortProperties=cm%3Amodified&sortAscending=false&sortAscending=false&propertyFilter=-all-&`,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This line has a length of 164. Maximum allowed is 120.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4791331)